coachmark/src/main/java/com/pseudoankit/coachmark/overlay/OverlayLayout.kt

```diff
@@ -0,0 +1,147 @@
+package com.pseudoankit.coachmark.overlay
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.UiComposable
+import androidx.compose.ui.layout.Layout
+import androidx.compose.ui.layout.Measurable
+import androidx.compose.ui.layout.Placeable
+import androidx.compose.ui.layout.layoutId
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.pseudoankit.coachmark.model.ToolTipPlacement
+import com.pseudoankit.coachmark.model.TooltipConfig
+
+/** Containing composable must use these values for layoutId on current and previous tooltip. */
+public enum class OverlayChildLayoutId { CURRENT, PREVIOUS }
+
+/** Proposed minimum useful tooltip width; this can be adjusted if needed */
+private const val TOOLTIP_MAX_WIDTH_OVERRIDE_PX = 30
+
+/**
+ * Extracted from DimOverlayEffect, in case another overlay effect needs it in the future.
+ *
+ * @param configCurrent of coach mark highlight; null suits case where there's no "current" tooltip in content
+ * @param configPrevious of coach mark highlight; null suits case where there's no "previous" tooltip in content
+ * @param gapTooltipScreen min distance between tooltip and edge of screen
+ */
+@Composable
+public fun OverlayLayout(
+    content: @Composable @UiComposable () -> Unit,
+    configCurrent: TooltipConfig?,
+    configPrevious: TooltipConfig?,
+    modifier: Modifier = Modifier,
+    gapTooltipScreen: Dp = 8.dp,
```

Comment:
can we rename, minDistanceFromScreen ? or any better suggestions if you have

Reply:
can we move 8.dp to CoachMarkDefaults.ToolTip

Reply:
How about just "padding", with the default on CoachMarkDefaults.Overlay.padding?

Reply:
it will be misleading bcz it's not padding right?
minOffset may be

Reply:
It's the distance between overlay edge and overlay content.  That's padding, right?  Like when a Row has padding within its background.

Reply:
On the other hand, users might not understand that tooltips are children of the overlay.  Plus the gap has no effect on the highlighted area.

Reply:
-- gapTooltipScreen min distance between tooltip and edge of screen

I was suggesting name based on your comment

Reply:
I think offset can be a good choice
`paddingForTooltip` should be good choice too I think